kola/tests/misc/cloudinit.go

```diff
@@ -78,3 +208,31 @@ func CloudInitScript(c cluster.TestCluster) {
 		c.Fatalf("userdata script produced unexpected value %q", out)
 	}
 }
+
+func CloudInitMultipartMime(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	expectKey := "AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEftQIHTRvUmyDCN7VGve4srz03Jmq6rPnqq+XMHMQUIL9c/b0l7B5tWfQvQecKyLte94HOPzAyMJlktWTVGQnY="
+
+	// Test that the hostname was set by the first multipart mime part that declares the "hostname"
+	// cloud-config option. The second one at the end should be ignored.
+	out := c.MustSSH(m, "hostnamectl")
+	if !strings.Contains(string(out), "Static hostname: example") {
+		c.Fatalf("hostname wasn't set correctly:\n%s", out)
+	}
+
+	// we can ignore the output. If the command fails, MustSSH will fail the test.
+	c.MustSSH(m, fmt.Sprintf("grep %s ~core/.ssh/authorized_keys", expectKey))
+
+	out = c.MustSSH(m, "ls -l /tmp/kola_*| wc -l")
+	if string(strings.Replace(string(out), "\n", "", -1)) != "7" {
+		c.Fatalf("expected 7 files in /tmp, found %q", out)
+	}
+
+	// All files should have the same content (42). These files should have been created by the cloud-config part
+	// that declares the write_files option.
+	c.MustSSH(m, `for f in $(ls /tmp/kola_*); do OUT=$(cat $f); if [ "$OUT" != 42 ]; then exit 1; fi; done`)
```

Comment:
```suggestion
	c.MustSSH(m, `for f in /tmp/kola_*; do OUT=$(cat $f); if [ "$OUT" != 42 ]; then exit 1; fi; done`)
```
If we expect the files to exist we can use this and if they don't, we get the `kola_*` filename and the test will fail as well.